Sources/Tracking/Util/DeviceInfo.swift

```diff
@@ -66,8 +66,22 @@ public class CIODeviceInfo: DeviceInfo {
         SdkVersion.version
     }
 
+    // Requirements:
+    // 1. Use - instead of _
+    // 2. We prefer to get the OS language that the user has set. First try to return that. If that does not succeed
+    // we default to the language set for the host app. The language returned for that will only return languagues
+    // that the app supports. If OS set to "es" but app does not support Spanish, then "es" will not be returned.
     public var deviceLocale: String {
-        Locale.current.identifier.replacingOccurrences(of: "_", with: "-")
+        if let osSetLanguage = Locale.preferredLanguages.first {
+            let locale = Locale(identifier: osSetLanguage)
+
+            if let languageCode = locale.languageCode,
```

Comment:
could this be simplified to
```
return locale.identifier.replacingOccurrences(of: "_", with: "-")
or
return osSetLanguage.replacingOccurrences(of: "_", with: "-")
```

Reply:
Looking at the code, I don't currently see an easy way to do that. There are so many possibilities for null to happen that's why this code has 3 `if let` statements.